src/sinks/datadog/logs/sink.rs

```diff
@@ -145,15 +133,16 @@ impl crate::sinks::util::encoding::Encoder<Vec<Event>> for JsonEncoding {
                 .expect("message is required (make sure the \"message\" semantic meaning is set)");
             log.rename_key(message_path.as_str(), event_path!("message"));
 
-            let host_path = log
-                .host_path()
-                .expect("message is required (make sure the \"host\" semantic meaning is set)");
-            log.rename_key(host_path.as_str(), event_path!("host"));
+            if let Some(host_path) = log.host_path() {
+                log.rename_key(host_path.as_str(), event_path!("hostname"));
+            }
 
-            if let Some(Value::Timestamp(ts)) = log
+            if let Some(Value::Timestamp(ts)) = log.remove(
+                log
                 .timestamp_path()
-                .and_then(|path| log.remove(path.as_str()))
-            {
+                .expect("timestamp is required (make sure the \"timestamp\" semantic meaning is set)")
```

Comment:
Is this something that might trip up users not expecting it? 🤔

Reply:
That's what Sink requirements is for. This will only fail if the new "log namespacing" feature is enabled _and_ there is no semantic meaning set for `timestamp`. However, the "log namespacing" feature will enable sink requirements automatically, which will ensure at startup that `timestamp` exists.

Reply:
🙌